files/brews/hub.rb

```diff
@@ -0,0 +1,26 @@
+require 'formula'
+
+class Hub < Formula
+  homepage "http://hub.github.com/"
+
+  if RUBY_PLATFORM =~ /x86_64/
+    url "https://github.com/github/hub/releases/download/v2.2.0-rc1/hub_mac_amd64_v2.2.0-rc1.tar.gz"
+    sha1 "413ce80d32f6313cef010e0edb1e664e1fff7427"
+  else
+    url "https://github.com/github/hub/releases/download/v2.2.0-rc1/hub_mac_386_v2.2.0-rc1.tar.gz"
+    sha1 "059e6d1759b568fd79bacfc397de199135056ff1"
+  end
+
+  def install
+    bin.install "hub"
+    man1.install Dir["man/*"]
+    bash_completion.install "etc/hub.bash_completion.sh"
+    zsh_completion.install "etc/hub.zsh_completion" => "_hub"
+  end
+
+  test do
+    HOMEBREW_REPOSITORY.cd do
+      assert_equal "bin/brew", shell_output("#{bin}/hub ls-files -- bin").strip
```

Comment:
Instead of checking version doing another meaningful Git command (ideally one that hits the API) would be great. But, as you say, this isn't Homebrew core so do whatever makes you happy :)